src/dht/provide.js

```diff
@@ -26,13 +25,22 @@ module.exports = configure(({ ky }) => {
     })
 
     for await (let message of ndjson(toIterable(res.body))) {
+      // 3 = QueryError
```

Comment:
Is it possible to extract some of this boilerplate
- `for await (let message of ndjson(toIterable(res.body)))`
- `if (message.Type === 3) `
- etc
into a function / class?